### DISCUSSION TITLE
Clear Browser when switching TabControl Tabs [WPF]

### DISCUSSION THREAD

**Discussion by tobyas** [cefsharp/CefSharp#4401] [General]
Hello,
I have a WPF Tab Control where Tab A contains the Browser and Tab B is empty. 
When switching from Tab A to Tab B the Browser is 'cleared' by calling Browser.LoadHtml("<html><head></head><body></body></html>");. 
When switching back from Tab B to Tab A the screen contains the old browser content for a short moment. Is there a way to prevent this / better way to clear the browser?

**Comment by amaitland** [comment]
> is 'cleared' by calling Browser.LoadHtml("");.

Calling `Load` is an async operation. When the browser becomes hidden the rendering is actually paused. So what you are seeing is not unexpected.

Each frame is rendered as a `Bitmap`. The following references should give you a better understanding of how rendering actually happens.

- https://github.com/cefsharp/CefSharp/blob/cefsharp/110/CefSharp.Wpf/Themes/Generic.xaml
- https://github.com/cefsharp/CefSharp/blob/cefsharp/110/CefSharp.Wpf/ChromiumWebBrowser.cs#L1007
- https://github.com/cefsharp/CefSharp/blob/cefsharp/110/CefSharp.Wpf/Rendering/WritableBitmapRenderHandler.cs#L52
- https://github.com/cefsharp/CefSharp/blob/cefsharp/110/CefSharp.Wpf/ChromiumWebBrowser.cs#L1821

There are hooks into the rendering if you want to implement a custom behaviour.
